src/Client.ts

```diff
@@ -5,22 +5,32 @@ import {
 } from '@kontent-ai/delivery-sdk';
 import packageInfo from '../package.json';
 import { selectedProjectCookieName } from './const';
-import { defaultProjectId } from './Utilities/SelectedProject';
+import validator from 'validator';
 
 const sourceTrackingHeaderName = 'X-KC-SOURCE';
 
-// environment variables
-const projectId = process.env.REACT_APP_PROJECT_ID || '';
 const previewApiKey = process.env.REACT_APP_PREVIEW_API_KEY || '';
 
 const cookies = new Cookies(document.cookie);
 
-let currentProjectId = projectId || cookies.get(selectedProjectCookieName);
-if (currentProjectId) {
-  cookies.set(selectedProjectCookieName, currentProjectId, { path: '/' });
-} else {
-  currentProjectId = defaultProjectId;
-}
+const getProjectId = (): string => {
+  const projectId = process.env.REACT_APP_PROJECT_ID || '';
+
+  let currentProjectId = projectId;
+
+  if (projectId) {
+    if (!validator.isUUID(projectId)) {
+      console.error(
+        `Your projectId (${projectId}) given in your environment variables is not a valid GUID.`
+      );
+    }
+    return projectId;
+  }
+  currentProjectId = cookies.get(selectedProjectCookieName);

```

Comment:
Why don't you also check the validity of the `projectId` from the cookie? It might also be invalid. :)